src/Controller/UserController.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\User;
+use App\Form\UserType;
+use App\Manager\UserManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class UserController extends AbstractController
+{
+    public function __construct(private UserManagerInterface $userManager)
+    {
+    }
+
+    #[Route('/users', name: 'user_list')]
+    public function listAction(): Response
+    {
+        return $this->render('user/list.html.twig', [
+            'controller_name' => 'UserController',
+            'users' => $this->userManager->manageListAction(),
+        ]);
+    }
+
+    #[Route('/users/create', name: 'user_create')]
+    public function createAction(Request $request): Response
+    {
+        $user = new User();
+        $form = $this->createForm(UserType::class, $user);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->userManager->manageCreateOrUpdate($user);
+            $this->addFlash('success', "L'utilisateur a été ajouté.");
+
+            return $this->redirectToRoute('user_list');
+        }
+
+        return $this->render('user/create.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    #[Route('/users/{id}/edit', name: 'user_edit')]
+    public function editAction(User $user, Request $request): Response
+    {
+        $form = $this->createForm(UserType::class, $user, [
+            'require_password' => false, 
+        ]);
+        $password = $user->getPassword();
```

Comment:
Tu ne vérifies pas que ton utilisateurs est authentifié ?

Reply:
en fait ici, tu veux récupérer le mot de passe de ton utilisateur authentifié ? dans ce cas il faut faire un `$this->getUer()->getPassword() ` et non le mot de passe de ton `$user` de ta fonction `ublic function editAction(User $user, Request $request)`. Je crois  que celui-ci n'est remplis qu'avec des informations modifiées par ton formulaire. donc ton password devrait être null ($user->getPassword()).

Reply:
https://symfony.com/doc/current/security.html#fetching-the-user-object

Reply:
Done